src/ios-scroll.js

```diff
@@ -0,0 +1,557 @@
+    if (this.bodyScrolled_) {
+      // If the user is still scrolling natively, this will stop it and start
+      // the synthetic scrolling.
+      event.preventDefault();
+      event.stopPropagation();
+
+      // The doc has scrolled natively, causing our synthetic scroll to be out
+      // of sync with the real scrollTop. We must calculate the difference
+      // and adjust.
+      const delta = -this.scrollMoveEl_./*OK*/getBoundingClientRect().top;
+      this.scrollTop_ += delta;
+      this.min_ -= delta;
+      this.max_ -= delta;
+      this.bodyScrolled_ = false;
+    }
+
+    this.setIframeInteractionAllowed_(false);
+  }
+
+  /**
+   * Called when the doc stops scrolling, either because there was not enough
+   * velocity to have synthetic momentum or we are at our target offset from
+   * the momentum.
+   */
+  scrollStopped_() {
+    // To allow native scrolling on iframes, we must adjust the doc's real
+    // scrollTop to match where we syntheticly scrolled (else, the user
+    // wouldn't be able to natively scroll up because Safari thinks
+    // they're "at" the top of the doc).
+    this.scrollTop_ += this.offset_;
+    this.min_ -= offset;
+    this.max_ -= offset;
+    this.scrollTo_(0);
+    // Suppress this body scrolled event, we expect it.
+    this.bodyScrolled_ = null;
+    this.scrollMoveEl_.style.transform = 'translateY(' + this.scrollTop_ + 'px)';
+    this.scrollMoveEl_./*OK*/scrollIntoView(true);
+
+    this.setIframeInteractionAllowed_(true);
+  }
+
+  /**
+   * Enables or disables interaction with iframes.
+   *
+   * @param {boolean} allowed
+   */
+  setIframeInteractionAllowed_(allowed) {
+    const value = allowed ? '' : 'none';
+    const styles = this.iframeStyles_;
+    for (let i = 0; i < styles.length; i++) {
+      styles[i].pointerEvents = value;
+    }
+  }
+
+  /**
+   * A callback bound to the `touchstart` event on the document.
+   *
+   * @param {TouchEvent} event
+   */
+  touchstart_(event) {
+    // We don't allow zooming
+    if (event.touches.length > 1) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    if (this.autoScrolling_) {
+      cancelAnimationFrame(this.autoScrolling_);
+      this.autoScrolling_ = 0;
+    } else {
+      this.scrollStarted_();
+    }
+
+    this.velocity_ = 0;
+    this.amplitude_ = 0;
+    this.timestamp_ = Date.now();
+    this.reference_ = this.eventY_(event);
+  }
+
+  /**
+   * A callback bound to the `touchmove` event on the document.
+   *
+   * @param {TouchEvent} event
+   */
+  touchmove_(event) {
+    // We don't allow zooming
+    if (event.touches.length > 1) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    const y = this.eventY_(event);
+    const delta = this.reference_ - y;
+    this.track_(delta);
+    this.reference_ = y;
+    this.scrollTo_(this.offset_ + delta);
+
+    event.preventDefault();
+    event.stopPropagation();
+    return false;
```

Comment:
does this affect things like carousel? I don't think it will. but would be good to check.